app/src/org/commcare/activities/FormHierarchyActivity.java

```diff
@@ -165,13 +181,18 @@ private void refreshView() {
         String hierarchyPath;
         try {
             hierarchyPath = FormHierarchyBuilder.populateHierarchyList(this, formList);
-        } catch (XPathTypeMismatchException e) {
+        } catch (XPathException e) {
             XPathErrorLogger.INSTANCE.logErrorToCurrentApp(e);
 
-            final String errorMsg = "Encounted xpath error: " + e.getMessage();
-            Toast.makeText(this, errorMsg, Toast.LENGTH_LONG).show();
-            setResult(RESULT_XPATH_ERROR);
```

Comment:
I removed this in favour of [this line](https://github.com/dimagi/commcare-android/blob/7a683f0ff101f3fd2a551ff2dcd44a0d6b7a4b45/app/src/org/commcare/views/UserfacingErrorHandling.java#L86), but [this code](https://github.com/dimagi/commcare-android/blob/7a683f0ff101f3fd2a551ff2dcd44a0d6b7a4b45/app/src/org/commcare/activities/FormEntryActivity.java#L355) suggests that the `RESULT_XPATH_ERROR` might still be needed, or I need to change it to `RESULT_CANCELED`?